spec/ios_l10n_helper_spec.rb

```diff
@@ -0,0 +1,162 @@
+require_relative './spec_helper'
+
+describe Fastlane::Helper::Ios::L10nHelper do
+  let(:test_data_dir) { File.join(File.dirname(__FILE__), 'test-data', 'translations', 'ios_l10n_helper') }
+
+  def fixture(name)
+    File.join(test_data_dir, name)
+  end
+
+  # Returns the `Encoding` instance guessed for that file. Useful to control that our fixtures are in the expected encoding when running tests.
+  def file_encoding(path)
+    File.read(path, mode: 'rb:BOM|UTF-8').encoding # Use BOM to determine encoding if present, fallback to UTF8
+  end
+
+  describe '#strings_file_type' do
+    it 'detects an XML-formatted strings file' do
+      xml_fixture = fixture('xml-format.strings')
+      expect(described_class.strings_file_type(path: xml_fixture)).to eq(:xml)
+    end
+
+    it 'detects a binary-plist strings file' do
+      bplist_fixture = fixture('bplist-format.strings')
+      expect(described_class.strings_file_type(path: bplist_fixture)).to eq(:binary)
+    end
+
+    it 'detects a text-formatted strings file' do
+      text_fixture = fixture('Localizable-utf16.strings')
+      expect(described_class.strings_file_type(path: text_fixture)).to eq(:text)
+    end
+
+    it 'returns nil on a non-existing path' do
+      expect(described_class.strings_file_type(path: '/invalid-path')).to be_nil
+    end
+
+    it 'returns nil on file that is not a strings file at all' do
+      png_fixture = fixture('not-a-strings-file.png')
+      expect(File).to exist(png_fixture)
+      expect(described_class.strings_file_type(path: '/invalid-path')).to be_nil
+    end
+  end
+
+  describe '#merge_strings' do
+    # Ensuring we test this against input files which are using different encodings (UTF-16BE & UTF-8) is an important case to cover
+    it 'properly merges 2 textual strings files with different encodings into a new one' do
+      paths = [fixture('Localizable-utf16.strings'), fixture('InfoPlist-utf8.strings')]
+
+      # Just making sure we don't accidentally change the encoding of the fixture files in the future
+      expect(file_encoding(paths[0])).to eq(Encoding::UTF_16BE)
+      expect(file_encoding(paths[1])).to eq(Encoding::UTF_8)
+
+      Dir.mktmpdir('a8c-release-toolkit-l10n-helper-tests-') do |tmp_dir|
+        output_file = File.join(tmp_dir, 'output.strings')
+        described_class.merge_strings(paths: paths, output_path: output_file)
+        expect(File.read(output_file)).to eq(File.read(fixture('expected-merged.strings')))
```

Comment:
Worth adding a check also for the output file encoding?

```suggestion
        expect(File.read(output_file)).to eq(File.read(fixture('expected-merged.strings')))
        expect(file_encoding(output_file)).to eq(Encoding::UTF_8)
```